tests/unit/conftest.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+from unittest.mock import MagicMock
+
+import pytest
+from ops.testing import Harness
+
+from charm import KubeflowProfilesOperator, KubernetesServicePatch
+
+
+@pytest.fixture
+def harness():
+    harness = Harness(KubeflowProfilesOperator)
+    # harness.set_leader(True)
+    harness.set_can_connect("kubeflow-profiles", True)
+    harness.set_can_connect("kubeflow-kfam", True)
+    return harness
+
+
+@pytest.fixture()
+def mocked_kubernetes_service_patcher(mocker):
+    mocker.patch.object(KubernetesServicePatch, "_namespace", lambda x, y: "")
+    mocker.patch.object(KubernetesServicePatch, "_patch", lambda x, y: None)
+
+    yield
+
+
+@pytest.fixture()
+def mocked_resource_handler(mocker):
+    mocked_resource_handler = mocker.patch("charm.KRH")
+    mocked_resource_handler.return_value = MagicMock()
+    yield mocked_resource_handler
```

Comment:
There is nothing wrong with this, but imo these are clearer if they're in the test_operator.py file rather than conftest.py.  This is just because anything in conftest.py is a bit hidden from view.  I don't think we have a team convention on where something like this would go though so changes aren't necessary if you prefer this way

Reply:
I think it's cleaner for the setup to be separated from the tests, although I see your point on clearance. perhaps add a docstring at the top of the test_operator.py to refer to conftest.py?

Reply:
If you're looking for cleanliness, what about storing them in a separate python file (helpers.py, fixtures,py, test_operator_fixtures.py, etc) and importing them from there?  The trouble with `conftest.py` is that it is an *automatically loaded* file.  It is essentially like an inherited global variable.  So now any other test executed from `/tests/unit` or any of it's subdirectories will always inherit your fixtures whether they want them or not.  

I'd also suggest that the lost clarity is worse than gained cleanliness.  Readers of `test_operator.py` no longer need to sift through these fixtures, but they're also not told where to find them if they want to know what they do.  They need to know to look in a conftest.py file and search for them.  Whereas if the fixtures are explicitly imported (like other package imports), there's a clear indicator about what's loaded and where to look at the source code

Reply:
great explanation, will move to test_operator_fixtures.py and import explicitly